probes/https-outbound-probe.js

```diff
@@ -0,0 +1,210 @@
+
+function HttpsOutboundProbe() {
+  Probe.call(this, 'https'); // match the name of the module we're instrumenting
+}
+util.inherits(HttpsOutboundProbe, Probe);
+
+HttpsOutboundProbe.prototype.attach = function(name, target) {
+  var that = this;
+  if (name == 'https') {
+    if (target.__outboundProbeAttached__) return target;
+    target.__outboundProbeAttached__ = true;
+
+    aspect.around(
+      target,
+      methods,
+      // Before 'https.request' function
+      function(obj, methodName, methodArgs, probeData) {
+        // Get HTTPS request method from options
+        var options = methodArgs[0];
+        var requestMethod = 'GET';
+        var urlRequested = '';
+        var headers = '';
+        if (typeof options === 'object') {
+          urlRequested = formatURL(options);
+          if (options.method) {
+            requestMethod = options.method;
+          }
+          if (options.headers) {
+            headers = options.headers;
+          }
+        } else if (typeof options === 'string') {
+          urlRequested = options;
+          var parsedOptions = url.parse(options);
+          if (parsedOptions.method) {
+            requestMethod = parsedOptions.method;
+          }
+          if (parsedOptions.headers) {
+            headers = parsedOptions.headers;
+          }
+        }
+
+        // Start metrics
+        that.metricsProbeStart(probeData, requestMethod, urlRequested);
+        that.requestProbeStart(probeData, requestMethod, urlRequested);
+
+        // End metrics
+        aspect.aroundCallback(
+          methodArgs,
+          probeData,
+          function(target, args, probeData) {
+            methodArgs.statusCode = args[0].statusCode;
+            that.metricsProbeEnd(probeData, requestMethod, urlRequested, args[0], headers);
+            that.requestProbeEnd(probeData, requestMethod, urlRequested, args[0], headers);
+          },
+          function(target, args, probeData, ret) {
+            // Don't need to do anything after the callback
+            return ret;
+          }
+        );
+      },
+      // After 'https.request' function returns
+      function(target, methodName, methodArgs, probeData, rc) {
+        // If no callback has been used then end the metrics after returning from the method instead
+        if (aspect.findCallbackArg(methodArgs) == undefined) {
+          // Need to get request method and URL again
+          var options = methodArgs[0];
+          var requestMethod = 'GET';
+          var urlRequested = '';
+          var headers = '';
+          if (typeof options === 'object') {
+            urlRequested = formatURL(options);
+            if (options.method) {
+              requestMethod = options.method;
+            }
+            if (options.headers) {
+              headers = options.headers;
+            }
+          } else if (typeof options === 'string') {
+            urlRequested = options;
+            var parsedOptions = url.parse(options);
```

Comment:
I rewrote this bit in both http-outbound-probe and https-outbound-probe. They're both now only calling `url.parse()` on metrics end.